tests/Dataset.test.ts

```diff
@@ -10,61 +10,68 @@ import { ANY_ERROR } from "./test_utils";
-    return async (inputUrl: string): Promise<T> => {
-      if (inputUrl !== url) {
-        throw new Error(`Input url '${inputUrl}' does not match expected url '${url}'.`);
-      }
-      return Promise.resolve(manifestJson);
-    };
+const DEFAULT_PATH = "https://some-path.json";
+
+const makeMockFetchMethod = <T>(url: string, manifestJson: T): ((url: string) => Promise<T>) => {
+  return async (inputUrl: string): Promise<T> => {
+    if (inputUrl !== url) {
+      throw new Error(`Input url '${inputUrl}' does not match expected url '${url}'.`);
+    }
+    return Promise.resolve(manifestJson);
   };
+};
 
-  class MockFrameLoader implements IFrameLoader {
-    width: number;
-    height: number;
+class MockFrameLoader implements IFrameLoader {
+  width: number;
+  height: number;
 
-    constructor(width: number = 1, height: number = 1) {
-      this.width = width;
-      this.height = height;
-    }
+  constructor(width: number = 1, height: number = 1) {
+    this.width = width;
+    this.height = height;
+  }
 
-    load(_url: string): Promise<Texture> {
-      return Promise.resolve(
-        new DataTexture(
-          new Uint8Array(this.width * this.height * 4),
-          this.width,
-          this.height,
-          RGBAFormat,
-          UnsignedByteType
-        )
-      );
-    }
+  load(_url: string): Promise<Texture> {
+    return Promise.resolve(
+      new DataTexture(
+        new Uint8Array(this.width * this.height * 4),
+        this.width,
+        this.height,
+        RGBAFormat,
+        UnsignedByteType
+      )
+    );
   }
+}
 
-  class MockArraySource implements ArraySource {
-    getBuffer<T extends FeatureDataType>(type: T): FeatureArrayType[T] {
-      return new featureTypeSpecs[type].ArrayConstructor([]);
-    }
-    getTexture(_type: FeatureDataType): Texture {
-      return new Texture();
-    }
-    getMin(): number {
-      return 0;
-    }
-    getMax(): number {
-      return 1;
-    }
+class MockArraySource implements ArraySource {
+  getBuffer<T extends FeatureDataType>(type: T): FeatureArrayType[T] {
+    return new featureTypeSpecs[type].ArrayConstructor([]);
   }
+  getTexture(_type: FeatureDataType): Texture {
+    return new Texture();
+  }
+  getMin(): number {
+    return 0;
+  }
+  getMax(): number {
+    return 1;
+  }
+}
 
-  class MockArrayLoader implements IArrayLoader {
-    load(_url: string): Promise<ArraySource> {
-      return Promise.resolve(new MockArraySource());
-    }
+class MockArrayLoader implements IArrayLoader {
+  load(_url: string): Promise<ArraySource> {
+    return Promise.resolve(new MockArraySource());
   }
+}
+
+export const makeMockDataset = async (manifest: AnyManifestFile): Promise<Dataset> => {
+  const dataset = new Dataset(DEFAULT_PATH, new MockFrameLoader(), new MockArrayLoader());
+  const mockFetch = makeMockFetchMethod(DEFAULT_PATH, manifest);
+  await dataset.open(mockFetch);
+  return dataset;
+};
```

Comment:
Moved outside the `describe` block to make this code more general, so it can be used for another test in `data_utils.ts`.